### PR TITLE
Add a sample for azure core http instrumentation

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

File: opentelemetry-api/java-agent/pom.xml [hunk 1]
```diff
@@ -17,6 +17,7 @@
       <module>extensions</module>
       <module>TelemetryFilteredBaseOnRequestDuration</module>
       <module>extensions/FilterSpanBasedOnDuration</module>
+        <module>TestAzureCore</module>
```
Comment (trask) [inline opentelemetry-api/java-agent/pom.xml:20]: maybe better name?
```suggestion
        <module>AzureHttpClient</module>
```

File: opentelemetry-api/java-agent/TestAzureCore/pom.xml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+<!--              <includes>-->
+<!--                <include>applicationinsights-agent-3.5.4-SNAPSHOT.jar</include>-->
+<!--              </includes>-->
+<!--            </resource>-->
+<!--          </resources>-->
+<!--        </configuration>-->
+<!--      </plugin>-->
+<!--    </plugins>-->
+<!--  </build>-->
+
```
Comment (trask) [inline opentelemetry-api/java-agent/TestAzureCore/pom.xml:83]: did you want to keep this commented out code?

**Review by trask — APPROVED** [review]

File: opentelemetry-api/java-agent/pom.xml (hunk 1 above)
Comment (heyams) [reply opentelemetry-api/java-agent/pom.xml:20]: will rename it. i will add a README too.

File: opentelemetry-api/java-agent/TestAzureCore/pom.xml (hunk 2 above)
Comment (heyams) [reply opentelemetry-api/java-agent/TestAzureCore/pom.xml]: yes, that is how we use a local copy of jar. i can reference it next time whenever needed.

Comment (heyams) [reply opentelemetry-api/java-agent/TestAzureCore/pom.xml]: it's intentional.

**Pull request merged**